File: go/config/Sntp.go
package config

import (
	"github.com/lvcdy/huawei-lte-api-go/session"
)

// Sntp 对应 config/Sntp.py。
type Sntp struct {
	*session.ApiGroup
}

// NewSntp 创建 Sntp API 分组。
func NewSntp(s *session.Session) *Sntp {
	return &Sntp{ApiGroup: session.NewApiGroup(s)}
}

// Config 对应 sntp/config.xml。
func (s *Sntp) Config() (map[string]interface{}, error) {
	return s.S.Get("sntp/config.xml", nil, "config")
}

File: src/config/sntp.rs
//! Sntp config group (`config/Sntp.py`).

use serde_json::Value;

use crate::api::get_value;
use crate::connection::Connection;
use crate::errors::Result;

/// Sntp config group.
pub struct SntpConfig<'a> {
    conn: &'a Connection,
}

impl<'a> SntpConfig<'a> {
    /// Build a group bound to `conn`.
    pub fn new(conn: &'a Connection) -> Self {
        SntpConfig { conn }
    }

    /// `sntp/config.xml`. SNTP configuration.
    pub fn config(&self) -> Result<Value> {
        get_value(self.conn, "config/sntp/config.xml")
    }
}
